pkg/http/handler/timeout.go

```diff
@@ -75,21 +82,35 @@ func (h *timeToFirstByteTimeoutHandler) ServeHTTP(w http.ResponseWriter, r *http
 		h.handler.ServeHTTP(tw, r.WithContext(ctx))
 	}()
 
-	timeout := getTimer(h.timeout)
-	var timeoutDrained bool
-	defer func() { putTimer(timeout, timeoutDrained) }()
+	firstByteTimeout := getTimer(h.firstByteTimeout)
+	var firstByteTimeoutDrained bool
+
+	idleTimeout := getTimer(h.idleTimeout)
+	var idleTimeoutDrained bool
```

Comment:
Since the idle timeout will (I think) be optional, I'd expect there to be a clause that'd excludes getting a timer for a 0 duration here.